plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -58,31 +57,30 @@
     private final String hostname;
 
     private final MetastoreSupportsDateStatistics metastoreSupportsDateStatistics = new MetastoreSupportsDateStatistics();
-    private final AtomicInteger chosenGetTableAlternative = new AtomicInteger(Integer.MAX_VALUE);
-    private final AtomicInteger chosenTableParamAlternative = new AtomicInteger(Integer.MAX_VALUE);
-    private final AtomicInteger chosenGetAllViewsAlternative = new AtomicInteger(Integer.MAX_VALUE);
-    private final AtomicInteger chosenAlterTransactionalTableAlternative = new AtomicInteger(Integer.MAX_VALUE);
-    private final AtomicInteger chosenAlterPartitionsAlternative = new AtomicInteger(Integer.MAX_VALUE);
+    private final ThriftHiveMetastoreIntrospection metastoreIntrospection;
 
     public DefaultThriftMetastoreClientFactory(
             Optional<SSLContext> sslContext,
             Optional<HostAndPort> socksProxy,
             Duration timeout,
             HiveMetastoreAuthentication metastoreAuthentication,
+            ThriftHiveMetastoreIntrospection metastoreIntrospection,
             String hostname)
     {
         this.sslContext = requireNonNull(sslContext, "sslContext is null");
         this.socksProxy = requireNonNull(socksProxy, "socksProxy is null");
         this.timeoutMillis = toIntExact(timeout.toMillis());
         this.metastoreAuthentication = requireNonNull(metastoreAuthentication, "metastoreAuthentication is null");
+        this.metastoreIntrospection = requireNonNull(metastoreIntrospection, "metastoreIntrospection is null");
         this.hostname = requireNonNull(hostname, "hostname is null");
     }
 
     @Inject
     public DefaultThriftMetastoreClientFactory(
             ThriftMetastoreConfig config,
             HiveMetastoreAuthentication metastoreAuthentication,
-            NodeManager nodeManager)
+            NodeManager nodeManager,
+            ThriftHiveMetastoreIntrospection metastoreIntrospection)
```

Comment:
There is no need to inject this.  The current code this replaces constructs `AtomicInteger` inline.